extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionFactoryProducer.java

```diff
@@ -5,20 +5,20 @@
 import jakarta.enterprise.inject.Typed;
 import jakarta.inject.Inject;
 import jakarta.persistence.EntityManagerFactory;
-import jakarta.persistence.PersistenceUnit;
 
 import org.hibernate.reactive.common.spi.Implementor;
 import org.hibernate.reactive.mutiny.Mutiny;
 import org.hibernate.reactive.mutiny.impl.MutinySessionFactoryImpl;
 
 import io.quarkus.arc.DefaultBean;
 import io.quarkus.arc.Unremovable;
+import io.quarkus.hibernate.orm.ReactivePersistenceUnit;
 import io.quarkus.hibernate.orm.runtime.JPAConfig;
 
 public class ReactiveSessionFactoryProducer {
 
     @Inject
-    @PersistenceUnit
+    @ReactivePersistenceUnit
     EntityManagerFactory emf;
```

Comment:
By the way, if you try to get rid of `@ReactivePersistenceUnit`, be aware that defining an actual CDI producer like this in extensions is actually an anti-pattern. We're expected to define synthetic beans, like we currently do for `EntityManagerFactory`. Maybe this would make it easier for you to _not_ register the "reactive" `EntityManagerFactory` as a bean. You'd probably have to copy/paste the CDI bean definition build steps from ORM though.